src/csvcubed/models/validatedmodel.py

```diff
@@ -0,0 +1,42 @@
+# this class will be class attributes and returns an array of errors
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List
+
+from csvcubedmodels.dataclassbase import DataClassBase
+from csvcubed.models.validationerror import ValidateModelProperiesError
+from csvcubed.utils.validations import (
+    validate_str_type,
+    validate_int_type,
+    validate_list,
+)
+
+ValidationFunction = Callable[[Any, str], List[ValidateModelProperiesError]]
+
+
+class ValidatedModel(DataClassBase):
+    def validate(self) -> List[ValidateModelProperiesError]:
+        validation_errors: List[ValidateModelProperiesError] = []
+        for (property_name, validation_function) in self._get_validations().items():
+            print(f"Validating '{property_name}'")
```

Comment:
Can you convert this to a `_logger.debug` call instead of print? And make sure to use the `%s` syntax for string building.